stock_putaway_product/models/product_putaway.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# © 2016 Jos De Graeve - Apertoso N.V. <Jos.DeGraeve@apertoso.be>
+# © 2016 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api, _
+
+
+class ProductPutawayStrategy(models.Model):
+    _inherit = 'product.putaway'
+
+    @api.model
+    def _get_putaway_options(self):
+        ret = super(ProductPutawayStrategy, self)._get_putaway_options()
+        return ret + [('per_product', 'Fixed per product location')]
+
+    product_location_ids = fields.One2many(
+        comodel_name='stock.product.putaway.strategy',
+        inverse_name='putaway_id',
+        string='Fixed per product location',
+        copy=True)
+    method = fields.Selection(
```

Comment:
Original field declared in old API with defined function _get_putaway_options to load selection options: https://github.com/odoo/odoo/blob/6d43f280c9bfa589a88dec9433948861a93b39d7/addons/stock/product.py#L542